firmware/examples/TaskScheduler/source/main.cpp

```diff
@@ -32,30 +44,27 @@ class PrinterTask : public rtos::Task<1024>
   };
 
  private:
+  inline static SemaphoreHandle_t print_mutex;
   const char * message_;
   uint32_t run_count_;
 };
 
+PrinterTask printer_one("Printer A", "I am a printer, I am faster");

```

Comment:
Can you put a comment here to tell the user, that they must be declared globally or be allocated on the heap using the `new` operator for this to work. Otherwise, the system will crash.